gvm/protocols/gmpv9/__init__.py

```diff
@@ -0,0 +1,454 @@
+            scanner_id=scanner_id,
+            usage_type=UsageType.SCAN,  # pylint: disable=W0212
+            function=self.create_task.__name__,
+            alterable=alterable,
+            hosts_ordering=hosts_ordering,
+            schedule_id=schedule_id,
+            alert_ids=alert_ids,
+            comment=comment,
+            schedule_periods=schedule_periods,
+            observers=observers,
+            preferences=preferences,
+        )
+
+    def create_tls_certificate(
+        self,
+        name: str,
+        certificate: str,
+        *,
+        comment: Optional[str] = None,
+        copy: Optional[str] = None,
+        trust: Optional[bool] = None
+    ) -> Any:
+        """Create a new TLS certificate
+
+        Arguments:
+            comment: Comment for the TLS certificate.
+            name: Name of the TLS certificate, defaulting to the MD5 fingerprint
+            copy: The UUID of an existing TLS certificate
+            trust: Whether the certificate is trusted.
+            certificate: The Base64 encoded certificate data (x.509 DER or PEM).
+
+        Returns:
+            The response. See :py:meth:`send_command`for details.
+        """
+        if not name:
+            raise RequiredArgument(
+                argument='name', function=self.create_tls_certificate.__name__
+            )
+        if not certificate:
+            raise RequiredArgument(
+                argument="certificate",
+                function=self.create_tls_certificate.__name__,
+            )
+
+        cmd = XmlCommand("create_tls_certificate")
+
+        if comment:
+            cmd.add_element("comment", comment)
+
+        if copy:
+            cmd.add_element("copy", copy)
+
+        cmd.add_element("name", name)
+        cmd.add_element("certificate", certificate)
+
+        if trust:
+            cmd.add_element("trust", _to_bool(trust))
+
+        return self._send_xml_command(cmd)
+
+    def get_tls_certificates(
+        self,
+        *,
+        filter: Optional[str] = None,
+        filter_id: Optional[str] = None,
+        include_certificate_data: Optional[bool] = None
+    ) -> Any:
+        """Request a list of TLS certificates
+
+        Arguments:
+            filter: Filter term to use for the query
+            filter_id: UUID of an existing filter to use for the query
+
+        Returns:
+            The response. See :py:meth:`send_command` for details.
+        """
+
+        cmd = XmlCommand("get_tls_certificates")
+
+        _add_filter(cmd, filter, filter_id)
+
+        if include_certificate_data is not None:
+            cmd.set_attribute(
+                "include_certificate_data", _to_bool(include_certificate_data)
+            )
+
+        return self._send_xml_command(cmd)
+
+    def modify_tls_certificate(
+        self,
+        tls_certificate_id: str,
+        *,
+        name: Optional[str] = None,
+        comment: Optional[str] = None,
+        copy: Optional[str] = None,
```

Comment:
Please drop the copy param from modify_tls_certificate and create an extra clone_tls_certificate method instead. Personally I don't like the design in the protocol api to pass a <copy> element to a modify command to create a clone. Therefore I moved clone into an extra method.
```suggestion
```

Reply:
But the `clone_tls_certificate` method will internally also call the `XmlCommand("modify_tls_certificate")` and from the user view it looks like a different call?
(I've not found a `clone_tls_certificate` command in the GMP Protocol)

Reply:
Nevermind. I just answered my own question by looking into comparable `clone_` methods.

Reply:
:grin: yes internally it is a modify_tls_certificate command. Some background for me the protocol is very confusing and expects different parameters depending on the specific parameters passed actually. And this behavior is **NOT** documented in detail. Therefore I tried to make things explicit and introduced new methods instead of additional parameters.